src/key/key.ts

```diff
@@ -23,10 +20,17 @@ export const generatePrivateKey = (): string => {
  * @returns 64 byte hexadecimal string of public key
  */
 export const getPublicFromPrivate = (priv: string): string => {
-    const key = secp256k1.keyFromPrivate(priv);
-    // Remove prefix "04" which represents it's uncompressed form.
-    return key
-        .getPublic()
-        .encode("hex")
-        .slice(2);
+    var enc = new TextEncoder();
+    return sodium.crypto_sign_ed25519_sk_to_pk(enc.encode(priv)).toString();
```

Comment:
I'm trying to make`getPublicFromPrivate()` function. To get the public key in `sodiumoxide`, the following process is being used:
```rust
    pub fn public_key(&self) -> PublicKey {
        let mut pk = PublicKey([0u8; PUBLICKEYBYTES]);
        unsafe {
            ffi::crypto_sign_ed25519_sk_to_pk(pk.0.as_mut_ptr(), self.0.as_ptr());
        }
        pk
    }
```
However, `libsodium-wrappers` do not seem to provide the `crypto_sign_ed25519_sk_to_pk()` function. In [this file](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/libsodium-wrappers/index.d.ts), we can check all the constant types and functions provided by `libsodium-wrappers`. But `crypto_sign_ed25519_sk_to_pk` is not there.

Reply:
Here is the wrapping symbol https://github.com/jedisct1/libsodium.js/blob/master/wrapper/symbols/crypto_sign_ed25519_sk_to_pk.json
The linked file is in Dfinity's repository.

Reply:
I found it too, so I modified it. However, it still doesn't work properly. Informations about errors are here:
```
UnhandledPromiseRejectionWarning: TypeError: sodium.crypto_sign_ed25519_sk_to_pk is not a function
UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict`
[DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Reply:
I read it but `libsodium-wrappers` does not expose the function. Let's use surmo version instead. It provides the function.